pkg/client/selfhosted/selfhosted.go

```diff
@@ -0,0 +1,232 @@
+package selfhosted
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/jetstack/version-checker/pkg/api"
+)
+
+const (
+	// This n=500 is a temporary work around until pagination is properly tested
+	// Not all versions support pagination AND not all Artifactory versions are handling the "latest" argument
+	lookupURL   = "%s/%s/%s/tags/list?n=500"
+	manifestURL = "%s/%s/%s/manifests/"
+)
+
+type Options struct {
+	URL      string
+	LoginURL string
+	Username string
+	Password string
+	Bearer   string
+}
+
+type Client struct {
+	*http.Client
+	Options
+}
+
+type AuthResponse struct {
+	Token string `json:"token"`
+}
+
+type TagResponse struct {
+	Name string   `json:"name"`
+	Tags []string `json:"tags"`
+}
+
+type Result struct {
+	Name      string  `json:"name"`
+	Timestamp string  `json:"last_updated"`
+	Images    []Image `json:"images"`
+}
+
+type Image struct {
+	Digest       string `json:"digest"`
+	OS           string `json:"os"`
+	Architecture string `json:"Architecture"`
+}
+
+type ManifestResponse struct {
+	SchemaVersion int        `json:"schemaVersion"`
+	MediaType     string     `json:"mediaType"`
+	Manifests     []Manifest `json:"manifests"`
```

Comment:
I think this is artifactory specific. The API, and the registry implementation puts the SHA in the http header `Docker-Content-Digest`. Hopefully artifactory does the same, else we will need to use the header as a fallback.

https://docs.docker.com/registry/spec/api/

Reply:
This is actually schema version v2: https://docs.docker.com/registry/spec/manifest-v2-2/

After some more testing I noticed however that not all API requests can deliver the v2 spec. To be honest this bums me out, API wise lol. 

I will make some changes to just use schema version 1 (not to confuse with API 1, its still v2 API but with a v1 scheme.)

Reply:
Implemented V2.1 now :)